dDatabase/CDClientDatabase/CDClientTables/CDMissionsTable.cpp

```diff
@@ -118,3 +118,25 @@ const CDMissions& CDMissionsTable::GetByMissionID(uint32_t missionID, bool& foun
 	return Default;
 }
 
+const std::set<int32_t> CDMissionsTable::GetMissionsForReward(LOT lot) {
+	std::set<int32_t> toReturn {};
+	for (const auto& entry : GetEntries()) {
```

Comment:
This seems expensive

Reply:
added a cache for this... it is indeed expensive, but this is also what the client does on every load of the UI for the achievement vendor, but instead we are just doing it on buy